src/consensus/aft/raft.h

```diff
@@ -1815,18 +1800,14 @@ namespace aft
         return;
       }
 
-      // When we force to become the primary we are going around the
-      // consensus protocol. This only happens when a node starts a new network
-      // and has a genesis or recovery tx as the last transaction
-      election_index = last_committable_index();
-
       // A newly elected leader must not advance the commit index until a
       // transaction in the new term commits. We achieve this by clearing our
-      // list committable indices - so nothing from a previous term is now
+      // list of committable indices - so nothing from a previous term is now
       // considered committable. Instead this new primary will shortly produce
       // their own signature, which _will_ be considered committable.
       committable_indices.clear();
 
+      const auto election_index = last_signature;
```

Comment:
For those following along at home (reading the comments): I removed this entirely, and the `rollback(election_index)` below, but a bunch of tests break because we stop tracking terms correctly. I think that this could logically be removed, but the `rollback()` call is setting some important other state (ie - its not actually ever rolling back, but it is telling every other system the current term + index). We can unpick that later, for now I've just demoted `election_index` from a member field to a local, and got it to also use `last_signature` so that `last_committable_index` can be removed.